diff --git a/src/cli.rs b/src/cli.rs
index 2b7a830..a0dfba1 100644
--- a/src/cli.rs
+++ b/src/cli.rs
@@ -138,10 +138,22 @@ pub(crate) fn handle_sub_commands(cli_args: ArgMatches) -> Result<SubCommandGive
                 .trim()
                 .to_string();
             let target_session = get_single_selection(sessions, Some("tmux capture-pane -ept {}"))?;
-            execute_tmux_command(&format!(
-                "tmux switch-client -t {}",
-                target_session.replace('.', "_")
-            ));
+
+            fn is_in_tmux_session() -> bool {
+                std::env::var("TERM_PROGRAM").is_ok_and(|program| program == "tmux")
+            }
+
+            if !is_in_tmux_session() {
+                execute_tmux_command(&format!(
+                    "tmux attach -t {}",
+                    target_session.replace('.', "_")
+                ));
+            } else {
+                execute_tmux_command(&format!(
+                    "tmux switch-client -t {}",
+                    target_session.replace('.', "_")
+                ));
+            }
 
             Ok(SubCommandGiven::Yes)
         }
