diff --git a/auto-side-windows.el b/auto-side-windows.el
index 73b2c67..8ea3f67 100644
--- a/auto-side-windows.el
+++ b/auto-side-windows.el
@@ -464,6 +464,10 @@ non-nil value to respect the display buffer actions defined by this package."
     (message "No side buffers.")))
 
 ;;;; Minor Mode
+  (defun auto-side-windows--managed-p (buffer action)
+    (memq (auto-side-windows--get-buffer-side buffer action)
+          '(top bottom left right)))
+
 ;;;###autoload
 (define-minor-mode auto-side-windows-mode
   "Toggle automatic side window management based on buffer rules.
@@ -474,9 +478,13 @@ provided functions to `display-buffer-alist` to enable this feature."
   :group 'auto-side-windows
   (if auto-side-windows-mode
       (add-to-list 'display-buffer-alist
-                   '(t auto-side-windows--display-buffer))
+                   (cons
+                    #'auto-side-windows--managed-p
+                    '(auto-side-windows--display-buffer)))
     (setq display-buffer-alist
-          (delete '(t auto-side-windows--display-buffer)
+          (delete (cons
+                   #'auto-side-windows--managed-p
+                   '(auto-side-windows--display-buffer))
                   display-buffer-alist))))
 
 (provide 'auto-side-windows)
